backend/utils/circuit_breaker.py

```
@@ -0,0 +1,171 @@
+    OPEN = "open"
+    HALF_OPEN = "half_open"
+
+
+class CircuitBreakerError(Exception):
+    """Raised when circuit breaker is open"""
+
+    pass
+
+
+import threading
+from typing import Optional, Union, Type
```

Ruff (0.11.9)

Redefinition of unused `Optional` from line 9

Remove definition: `Optional`

(F811)